### PR TITLE
fix for right hand function declaration, for prototypical extension

### PR DESCRIPTION
**Pull request by karelm**
Fix for Issue #8 - functions declared by prototypical extension are not commented

### CODE REVIEW

**Comment by karelm** [comment]
When I try to generate comments for functions declared by extending the prototype, smartcomments doesn't appear to add documentation.

```
function someFunction(x){
    return x;
}

someFunction.prototype.test = function(a){
    return a;
}
```

If you run this through the smartcomments generator, you get:

```
/**
 * Description
 * @method someFunction
 * @param {} x
 * @return x
 */
function someFunction(x){
    return x;
}

someFunction.prototype.test = function(a){
    return a;
}
```

but it should generate a comment for the test function as well

**Comment by smartcomments** [comment]
Thanks so much karelm, for your SmartComments contribution. We are reviewing your pullrequest right now.

File: templates/default.js [hunk 1]
```diff
@@ -143,6 +143,19 @@ var MyTemplate = {
                         comment_data.node = params.node;
                         instance.buildComment(comment_data);
                     }
+                    else if(parent.type === 'AssignmentExpression') {
+                        
+                        if(parent.right.type === 'FunctionExpression') {
+                            
+                            if(parent.right.id.name){
```
Comment (smartcomments) [inline templates/default.js]: There is a minor problem with the following 3 lines. The problem is that the function name is on the left side of the assignment node, The way to access to the function name is to use parent.left.property.name, not parent.right.id.name. I leave a screenshot showing all assignment node's properties. Everything else looks good. Please upload these changes as soon as possible to be able to merge his pullrequest. Thank you again.

bug

**Comment by karelm** [comment]
I hope this fixes the issue.

**Comment by smartcomments** [comment]
+1

**Comment by karelm** [comment]
Glad I could help. If you need any other help, I don't mind lending a hand.

**Comment by darielnoel** [comment]
Thank you very much,we begin with 0.2.0 version spring soon, and your help can be very useful. You have any email where we can contact? Sorry, I almost forgot, please enter http://smartcomments.github.io/ and search (Ctrl + F) "elm". Regards.

**Pull request merged**